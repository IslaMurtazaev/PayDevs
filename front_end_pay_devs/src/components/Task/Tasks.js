import React, { Component } from "react";
import { connect } from "react-redux";

import { tasklyActions } from "../../actions/task";
import TaskItem from "./TaskItem";
import CreateTask from "./CreateTask";

class Tasks extends Component {
  componentDidMount() {
    this.props.getAllTasks(this.props.projectId);
  }

  render() {
    const { tasks, projectId } = this.props;

    return (
      <div>
        {tasks.length > 0 && <h3 className="taskHeader">Your tasks</h3>}

        <div>
          {tasks.map(task => (
            <TaskItem
              key={task.id}
              task={task}
              onDelete={this.props.removeTask}
              projectId={projectId}
            />
          ))}
        </div>

        <hr />

        <CreateTask projectId={projectId} />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    tasks: state.tasks
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getAllTasks: projectId => {
      dispatch(tasklyActions.getAll(projectId));
    },
    removeTask: taskId => {
      dispatch(tasklyActions.remove(taskId));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Tasks);
