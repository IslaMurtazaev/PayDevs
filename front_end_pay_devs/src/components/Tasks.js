import React, { Component } from "react";
import { connect } from "react-redux";

import {tasklyActions} from '../actions/taskly';
import TaskItem from './TaskItem'



class Tasks extends Component {
  onClick() {
    this.props.onGetAllProjects();
  }

  componentDidMount() {
    const { project } = this.props;
    this.props.onGetAllTasks(project.id);
  }

  render() {
    const tasks = this.props.tasks
    const { project } = this.props;
    console.log(project)

    return (
      <div>
        <ul>
          {tasks.map(task=>
            <li key={task.id}>
                <TaskItem task={task} onDelete={this.props.onDeleteTasks} projectId={project.id}/>
            </li>

          )}
        </ul>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
    // let project = state.projects.find(
    //   product => product.id === Number(ownProps.match.params.id)
    // );
    
    return {
      tasks: state.tasks
    };
  };

const mapDispatchToProps = (dispatch) => {
    return {
        onGetAllTasks: (projectId) => {
            dispatch(tasklyActions.getAll(projectId))
        },
        onDeleteTasks: (taskId) => {
            dispatch(tasklyActions.taksDelete(taskId))
        }
    }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Tasks);
