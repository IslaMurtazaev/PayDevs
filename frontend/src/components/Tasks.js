import React, { Component } from "react";

import TaskItem from "./TaskItem";
import CreateTask from "./CreateTask";

class Tasks extends Component {
  componentWillMount() {
    let { tasks, projectId, getAllTasks } = this.props;
    if (!tasks.length || tasks[0].projectId !== projectId)
      getAllTasks(projectId);
  }

  render() {
    const { tasks, projectId, removeTask } = this.props;

    return (
      <div>
        {tasks.length > 0 && <h3 className="header">Your tasks</h3>}

        <div className="tasks">
          {tasks.map(task => (
            <TaskItem
              key={task.id}
              task={task}
              onDelete={removeTask}
              projectId={projectId}
            />
          ))}
        </div>

        <CreateTask projectId={projectId} />
      </div>
    );
  }
}

export default Tasks;
