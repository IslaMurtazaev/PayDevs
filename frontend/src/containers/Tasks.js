import { connect } from "react-redux";

import { tasklyActions } from "../actions/task";
import Tasks from "../components/Tasks";

const mapStateToProps = state => {
  return {
    tasks: state.tasks
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getAllTasks: projectId => {
      dispatch(tasklyActions.getAll(projectId));
    },
    removeTask: taskId => {
      dispatch(tasklyActions.remove(taskId));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Tasks);
