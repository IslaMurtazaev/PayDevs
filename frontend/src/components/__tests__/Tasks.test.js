import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
jest.mock("../../../index.js", () => require("history"));

import Tasks from "../../containers/Tasks";

describe("<Tasks />", () => {
  const initialState = {
    tasks: [
      { id: 1, title: "Write Unit Tests", description: "desc1", price: 15, paid: true, completed: true },
      { id: 2, title: "Write Integraion Tests", description: "desc2", price: 20, paid: true, completed: false },
      { id: 3, title: "Write UI Tests", description: "desc3", price: 22, paid: false, completed: true },
      { id: 3, title: "Present the project", description: "desc4", price: 30, paid: false, completed: false }
    ]
  };

  const mockStore = configureStore();
  const store = mockStore(initialState)

  const container = shallow(<Tasks store={store} />).dive()

  it("renders rateHeader if there are props.tasks", () => {
    expect(container.find(".taskHeader").text()).toBe("Your tasks");
  });
 
  it("matches previous snapshot", () => {
    expect(toJson(container)).toMatchSnapshot();
  })
});
