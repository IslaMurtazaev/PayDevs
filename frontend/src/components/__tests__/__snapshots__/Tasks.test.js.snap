// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tasks /> matches previous snapshot 1`] = `
<div>
  <h3
    className="taskHeader"
  >
    Your tasks
  </h3>
  <div
    className="tasks"
  >
    <TaskItem
      key="1"
      onDelete={[Function]}
      task={
        Object {
          "completed": true,
          "description": "desc1",
          "id": 1,
          "paid": true,
          "price": 15,
          "title": "Write Unit Tests",
        }
      }
    />
    <TaskItem
      key="2"
      onDelete={[Function]}
      task={
        Object {
          "completed": false,
          "description": "desc2",
          "id": 2,
          "paid": true,
          "price": 20,
          "title": "Write Integraion Tests",
        }
      }
    />
    <TaskItem
      key="3"
      onDelete={[Function]}
      task={
        Object {
          "completed": true,
          "description": "desc3",
          "id": 3,
          "paid": false,
          "price": 22,
          "title": "Write UI Tests",
        }
      }
    />
    <TaskItem
      key="3"
      onDelete={[Function]}
      task={
        Object {
          "completed": false,
          "description": "desc4",
          "id": 3,
          "paid": false,
          "price": 30,
          "title": "Present the project",
        }
      }
    />
  </div>
  <Connect(CreateTaskForm) />
</div>
`;
